insert_string.c

```diff
@@ -8,18 +8,14 @@
 #include "zbuild.h"
 #include "deflate.h"
 
-/* ===========================================================================
- * Update a hash value with the given input byte
- * IN  assertion: all calls to to UPDATE_HASH are made with consecutive
- *    input characters, so that a running hash key can be computed from the
- *    previous key instead of complete recalculation each time.
- */
-#define HASH_SLIDE 16  // Number of bits to slide hash
+#define HASH_SLIDE           16
 
-#define UPDATE_HASH(s, h, val) \
-    h = ((val * 2654435761U) >> HASH_SLIDE);
+#define HASH_CALC(s, h, val) h = ((val * 2654435761U) >> HASH_SLIDE);
+#define HASH_CALC_VAR        h
+#define HASH_CALC_VAR_INIT   uint32_t h = 0
 
-#define INSERT_STRING       insert_string_c
-#define QUICK_INSERT_STRING quick_insert_string_c
+#define UPDATE_HASH          update_hash_c
```

Comment:
I wonder if I should change this to use something like `#define ISUFFIX(func) func##_c` like in my other PR #987.

Reply:
We should always aim for consistency... Otherwise the code would look like amateurs copy-pasted code from random sources.

Reply:
Actually I think I will leave it as is and if we need to refactor later (in both chunkset and insert_string) I can.